### PR TITLE
Implemented Change password flow

### PR DESCRIPTION
**Pull request by ebernevec**

### CODE REVIEW

File: src/main/java/com/endava/soa_5/actions/ChangePasswordActions.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+import static com.endava.soa_5.tools.Constants.PROPERTY_CHANGE_PASSWORD;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+public class ChangePasswordActions extends LogInAction {
+    static Logger log = LoggerFactory.getLogger(ChangePasswordActions.class);
+    ChangePasswordObjects changePasswordObjects = new ChangePasswordObjects();
+    PropertyReader propertyChangePassword = new PropertyReader(PROPERTY_CHANGE_PASSWORD);
+    WebDriverWait wait = new WebDriverWait(driver, 10);
```
Comment (MarcelMorozniuc) [inline src/main/java/com/endava/soa_5/actions/ChangePasswordActions.java]: Can be deleted if use wait method from GenericActions

File: src/main/java/com/endava/soa_5/actions/ChangePasswordActions.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    LogInPageObject logInPageObject = new LogInPageObject();
+    String newPassword = getNewPassword();
+
+    public void userLogsIn() {
+        logInPageObject.getLogInLink().click();
```
Comment (MarcelMorozniuc) [inline src/main/java/com/endava/soa_5/actions/ChangePasswordActions.java:27]: Better to have waiter for elements

File: src/main/java/com/endava/soa_5/actions/ChangePasswordActions.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        logInPageObject.getEmailInputField().sendKeys(propertyChangePassword.getPropertyValue("passUserName"));
+        logInPageObject.getPasswordInputField().sendKeys(propertyChangePassword.getPropertyValue("currentPassword"));
+        logInPageObject.getLogInButton().click();
+    }
+
+    public void navigateToChangePasswordPage() {
+        wait.until(ExpectedConditions.visibilityOf(changePasswordObjects.getUserEmail()));
```
Comment (MarcelMorozniuc) [inline src/main/java/com/endava/soa_5/actions/ChangePasswordActions.java]: Why don't use wait method from GenericActions ?
Can be used as well in next steps

File: src/main/java/com/endava/soa_5/actions/ChangePasswordActions.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        logInPageObject.getPasswordInputField().sendKeys(newPassword);
+        logInPageObject.getLogInButton().click();
+    }
+
+    public void checkThatUserIsLoggedIn() {
+        assertThat("LogIn was successful", checkAccountNameIsDisplayed(), is(propertyChangePassword.getPropertyValue("passUserName")));
+    }
+
+    public void writePass(String value) throws IOException {
+        Path filePath = Paths.get("src/main/resources/configuration/ChangePasswordCredentials.properties");
```
Comment (MarcelMorozniuc) [inline src/main/java/com/endava/soa_5/actions/ChangePasswordActions.java:86]: Can be simplified by adding in PropertyReader another method to set property (setProperty()), but it's up to you

**Review by NadejdaClima — APPROVED** [review]

**Pull request merged**